src/strings.bif

```diff
@@ -64,7 +64,15 @@ function string_cat%(...%): string
 	%{
 	int n = 0;
 	for ( const auto& a : @ARG@ )
+		{
+		if ( a->GetType()->Tag() != TYPE_STRING )
+			{
+			zeek::reporter->Error("string_cat() requires string arguments");
+			return zeek::make_intrusive<zeek::StringVal>("");
```

Comment:
```suggestion
			return val_mgr->EmptyString();
```

Otherwise might be helpful to return the violating pos/type, but no strong opinion.

Reply:
Some of the error messages in BIFs return just a string (via a call to `Reporter::Error`) and some of them report a location (via `emit_builtin_error`). I have no idea why one or the other is used in each case.